### PR TITLE
Manage queue configuration for Linux audit

### PR DESCRIPTION
**Pull request by theopolis**
This adds 3 controls for Linux audit:
- The `backlog_wait_time`, the Hz to pause each enqueue within the kernel.
- The `blacklog_limit`, the size of the queue.
- The failure action.

When osquery terminates safely the default settings for audit are restored.

### CODE REVIEW

File: osquery/events/linux/audit.cpp [hunk 1]
```diff
@@ -85,6 +85,9 @@ Status AuditEventPublisher::setUp() {
     // Want to set a min sane buffer and maximum number of events/second min.
     // This is normally controlled through the audit config, but we must
     // enforce sane minimums: -b 8192 -e 100
+    audit_set_backlog_wait_time(handle_, 1);
+    audit_set_backlog_limit(handle_, 256);
+    audit_set_failure(handle_, AUDIT_FAIL_SILENT);
```
Comment (theopolis) [inline osquery/events/linux/audit.cpp]: We may want to increase this queue size.

**Comment by osqueryer** [comment]
:-1: The commit cd0cfbc76d84593adebe49b52e8e5e146c960d5e (Job results: [3784](https://jenkins.osquery.io/job/osqueryPullRequestLinux/3784/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:no_good: The commit 6f116f4c5173e3a887bc51dd463f32d902568e83 (Job results: [768](https://jenkins.osquery.io/job/osqueryPullRequestAudit/768/)) failed the code audit or documentation test.

**Comment by theopolis** [comment]
This needs an end-2-end benchmark test from the Audit publisher to the `process_events` subscriber.

Here's an example input to stress:
```
(t,a) 1307 cwd="/home/reed/git/github/osquery"
(t,a) 1302 item=0 name="/usr/sbin/python" nametype=UNKNOWN
(t,a) 1327 proctitle=2F7573722F62696E2F656E7600707974686F6E002E2F626F6D622E70790039
(t,a) 1300 arch=c000003e syscall=59 success=yes exit=0 a0=7ffd7448861f a1=7ffd744887d0 a2=7ffd744887f0 a3=5b3 items=2 ppid=28423 pid=28857 auid=4294967295 uid=1000 gid=1000 euid=1000 suid=1000 fsuid=1000 egid=1000 sgid=1000 fsgid=1000 tty=pts20 ses=4294967295 comm="python" exe="/usr/bin/python2.7" key=(null)
(t,a) 1309 argc=3 a0="python" a1="./bomb.py" a2="9"
```

**Comment by osqueryer** [comment]
:-1: The commit e54b40380886510e070a8597a59f3b2839d20203 (Job results: [3789](https://jenkins.osquery.io/job/osqueryPullRequestLinux/3789/)) failed one or more tests (**Linux**).

**Pull request merged**